src/main/java/slimeknights/tconstruct/plugin/jei/ToolBuildingCategory.java

```diff
@@ -54,18 +55,21 @@ public void setRecipe(IRecipeLayoutBuilder builder, ToolBuildingRecipe recipe, I
       recipe.getExtraRequirements().stream().map(ingredient -> Arrays.asList(ingredient.getItems()))).toList();
     List<LayoutSlot> layoutSlots = recipe.getLayoutSlots();
 
+    if (layoutSlots.isEmpty()) {
+      // fallback to tinker station or anvil
+      layoutSlots = StationSlotLayoutLoader.getInstance().get(TConstruct.getResource(recipe.requiresAnvil() ? "tinkers_anvil" : "tinker_station")).getInputSlots();
+    }
```

Comment:
I kinda want to leave this up to the recipe to handle. Perhaps instead of (or in addition to) `getSlotLayoutId` we have a method like `getLayoutSlots` which also caches the list in the recipe instance? Advantage of a getter there is you could subclass the recipe and swap out the default if you wish for a more complicated tool recipe.